Author: Scott Moser <smoser@ubuntu.com>
Bug: https://launchpad.net/bugs/1353008
Applied-Upstream: yes revno 1012
Description: cloud-init-local depends on /run. reflect that in upstart job.
 With the writing of cloud-init status, cloud-init-local needs to
 have /run mounted.  The issue we were seeing was a race where:
  cloud-init-local creates /run/cloud-init
  /run is mounted
  cloud-init-local tries to link a file into /run/cloud-init
 .
 that directory was no longer visisable as /run was mounted over
 the top.
 .
 This also fixes a bug that would occur if a local datasource
 ran and found a source before cloud-init-nonet ran and networking
 had not yet come up.  If that occurred then cloud-init-nonet would
 exit without blocking network.
--- a/upstart/cloud-init-local.conf
+++ b/upstart/cloud-init-local.conf
@@ -1,6 +1,6 @@
 # cloud-init - the initial cloud-init job
 #   crawls metadata service, emits cloud-config
-start on mounted MOUNTPOINT=/
+start on mounted MOUNTPOINT=/ and mounted MOUNTPOINT=/run
 
 task
 
--- a/upstart/cloud-init-nonet.conf
+++ b/upstart/cloud-init-nonet.conf
@@ -58,10 +58,6 @@ script
    # static_network_up already occurred
    static_network_up && exit 0
 
-   # obj.pkl comes from cloud-init-local (or previous boot and
-   # manual_cache_clean)
-   [ -f /var/lib/cloud/instance/obj.pkl ] && exit 0
-
    dowait 10
    dowait 120
    msg "gave up waiting for a network device."
